tests/lib/rules/no-unused-prop-types.js

```diff
@@ -2480,6 +2480,22 @@ ruleTester.run('no-unused-prop-types', rule, {
         }
         MyComponent.propTypes = { * other() {} };
       `
+    },
+    {
+      code: `
+        class MyComponent extends React.Component {
+          componentDidMount() {
+            this.helperMethod();
+          }
+          componentWillReceiveProps(nextProps) {
+            this.helperMethod(nextProps);
+          }
+          helperMethod(props = this.props) {
```

Comment:
This should work. Passing around the entire props object is a massive antipattern anyways, but in this case, there’s no guarantee that nobody isnt calling it at runtime with “not a props object”.

The correct and only solution here is to destructured the props first, and pass individual props into the methods.